components/omega/doc/design/DataTypes.md

```diff
@@ -0,0 +1,121 @@
+<!--- DataTypes Requirements and Design --------------------------------------->
+
+# OMEGA Requirements and Design:
+
+# *DataTypes*
+
+
+## 1 Overview
+
+Data type aliases are a useful means for managing precision within OMEGA and
+to improve readability for for YAKL array data types. This header file 
+defines a number of aliases to common data types used throughout OMEGA.
+
+
+## 2 Requirements
+
+
+### 2.1 Requirement: Both flexible and fixed floating point
+
+We wish to explore mixed precision and lower precision floating point within
+OMEGA so require both a generic floating point type that can be switched at 
+compile time as well as explicit single and double-precision (32-bit, 64-bit) 
+types to enforce precision where needed.  
+
+### 2.2 Desired: Readability for array types
+
+OMEGA will be using YAKL array data types which have a long syntax. For
+readability, these should be aliased to a shorter, intuitive data type.
+
+### 2.3 Desired: Fixed width integer types
+
+For better control over integer storage, it might be better to define
+fixed-width integer data types rather than the generic int, long.
+
+### 2.4 Future: Floating point conversion
+
+Currently, language standard floating point conversion should be adequate,
+but future mixed-precision implementations may require more reproducible
+methods for implementation-dependent conversion. 
+
+## 3 Algorithmic Formulation
+
+No algorithms needed yet, though reproducibility in type conversion may
+require reproducible algorithms in the future.
+
+## 4 Design
+
+Currently, the data types can be defined in a single header file that sets
+up aliases.
+
+### 4.1 Data types and parameters
+
+#### 4.1.1 Parameters 
+
+We will use a global cpp define -D SINGLE_PRECISION at compile time as a 
+global switch for using single precision as the default real data type. 
+Otherwise, the default real will be double precision. 
```

Comment:
@grnydawn  This is a good idea, but suspect it may be problematic trying to do it with a CPP switch and global include like this. For example, managing the case where argument lists still need to match the rest of the code while vars internal to the routine can be converted. So I was assuming these tests would need some manual intervention anyway. It's good to keep this in mind though if we can make it work.